File: 2021/day02.rb
input = File.read("/Users/hhh/JungleGym/advent_of_code/2021/inputs/day02.in").split("\n")

x, y = 0, 0
x2, y2, a = 0, 0, 0

input.each do |i|
  dir, d = i.split " "
  d = d.to_i
  case dir
  when "forward"
    x += d
    x2 += d
    y2 += a*d
  when "down"
    y += d
    a += d
  when "up"
    y -= d
    a -= d
  end
end

p "Part 1: #{x*y}"
p "Part 2: #{x2*y2}"

File: 2021/go/day02/day02.go
package main

import(
	"bufio"
	"os"
	"fmt"
	"strings"
	"strconv"
)

func main() {
	file, err := os.Open("/Users/hhh/JungleGym/advent_of_code/2021/inputs/day02.in")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	s := bufio.NewScanner(file)
	x, y := 0,0
	for s.Scan() {
		res := strings.Split(s.Text(), " ")
		dir := res[0]
		d, _ := strconv.Atoi(res[1])
		switch dir {
			case "forward":
				x += d
			case "down":
				y += d
			case "up":
				y -= d
		}
	}
	fmt.Printf("Day 1: %d\n", x*y)
}
